### PR TITLE
tests(otelgin): move test files and trim module dependencies

### PR DESCRIPTION
**Pull request by flc1125**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Added support for providing `endpoint`, `pollingIntervalMs` and `initialSamplingRate` using environment variable `OTEL_TRACES_SAMPLER_ARG` in `go.opentelemetry.io/contrib/samples/jaegerremote`. (#6310)
 - Added support exporting logs via OTLP over gRPC in `go.opentelemetry.io/contrib/config`. (#6340)
 
+### Changed
+
+- Move the test files of `go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin` to the tests directory and simplify the dependencies of the main package. (#6360)
```
Comment (dmathieu) [inline CHANGELOG.md:18]: This change isn't user-facing. So it doesn't need a changelog entry.

Comment (flc1125) [reply CHANGELOG.md]: It has been processed, but it needs to be troublesome to add the tag of skipchangelog

Comment (dmathieu) [reply CHANGELOG.md]: I added the label.

**Review by dmathieu — APPROVED** [review]
Not having b3 as a dependency of the instrumentation anymore is nice.

**Review by pellared — APPROVED** [review]

File: instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go [hunk 2]
```diff
@@ -10,13 +10,14 @@ import (
...
 
-	"go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin/internal/semconvutil"
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/codes"
 	"go.opentelemetry.io/otel/propagation"
 	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
 	oteltrace "go.opentelemetry.io/otel/trace"
+
+	"go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin/internal/semconvutil"
```
Comment (dmathieu) [inline instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go:20]: Same comment as with the other PRs. Could you keep the import order the same?

**Pull request merged**